Sources/GDBRemote/DebugSessionImpl.cpp

```diff
@@ -178,55 +178,45 @@ ErrorCode DebugSessionImpl::queryStopCode(Session &session,
     return kErrorProcessNotFound;
   }
 
-  bool readRegisters = true;
-  StopInfo const &info = thread->stopInfo();
-
-  Architecture::CPUState state;
+  // Directly copy the fields that are common between ds2::StopInfo and
+  // ds2::GDBRemote::StopInfo.
+  static_cast<ds2::StopInfo &>(stop) = thread->stopInfo();
```

Comment:
That's not a bad idea. I'd have to do an assignment operator though, not a
copy constructor here, because `stop` is already constructed.

On Wed, Mar 30, 2016 at 12:27 PM, strager <notifications@github.com> wrote:

> In Sources/GDBRemote/DebugSessionImpl.cpp
> <https://github.com/facebook/ds2/pull/240#discussion_r57948427>:
>
> > @@ -178,55 +178,45 @@ ErrorCode DebugSessionImpl::queryStopCode(Session &session,
> >      return kErrorProcessNotFound;
> >    }
> >
> > -  bool readRegisters = true;
> > -  StopInfo const &info = thread->stopInfo();
> > -
> > -  Architecture::CPUState state;
> > +  // Directly copy the fields that are common between ds2::StopInfo and
> > +  // ds2::GDBRemote::StopInfo.
> > +  static_cast<ds2::StopInfo &>(stop) = thread->stopInfo();
>
> Make a StopInfo(const ds2::StopInfo &) "copy" constructor instead?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/240/files/84cd7a36b73b6e0a17d8aea18de39fdd0010de01..f447e66aff19c96eea35085a191440aeb1ed6234#r57948427>
>



-- 
Stephane Sezer